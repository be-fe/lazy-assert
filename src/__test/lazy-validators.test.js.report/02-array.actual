{}
  01 or : "123", string | number : {}
    result : true
  01 or : 123, string | number : {}
    result : true
  01 or : null, string | number : {}
    message : "Did not match or-array"
    reason : "all-types-failed"
    result : false
    subResult : []
      0 : {}
        message : "Target value is not of the expected type"
        reason : "string"
        result : false
        type : "type"
        vIndex : 0
      1 : {}
        message : "Target value is not of the expected type"
        reason : "number"
        result : false
        type : "type"
        vIndex : 1
    type : "or-array"
  02 and : false, func error : {}
    message : "Did not meet all validators in and array"
    reason : "some-failed"
    result : false
    subResult : []
      0 : {}
        funcInfo : undefined
        message : "Target did not pass the function check"
        reason : "returned-false"
        result : false
        type : "function"
        vIndex : 2
    type : "and-array"
  02 and : false, type error : {}
    message : "Did not meet all validators in and array"
    reason : "some-failed"
    result : false
    subResult : []
      0 : {}
        message : "Target value is not of the expected type"
        reason : "number"
        result : false
        type : "type"
        vIndex : 1
    type : "and-array"
  02 and : true : {}
    result : true
  03 > : false : {}
    funcInfo : "Comparison : > 3"
    message : "Target did not pass the function check"
    reason : "returned-false"
    result : false
    type : "function"
  03 > : true : {}
    result : true
  04 < : false : {}
    funcInfo : "Comparison : < 3"
    message : "Target did not pass the function check"
    reason : "returned-false"
    result : false
    type : "function"
  04 < : true : {}
    result : true
  05 >= : false : {}
    funcInfo : "Comparison : >= 3"
    message : "Target did not pass the function check"
    reason : "returned-false"
    result : false
    type : "function"
  05 >= : true : {}
    result : true
  06 <= : false : {}
    funcInfo : "Comparison : <= 3"
    message : "Target did not pass the function check"
    reason : "returned-false"
    result : false
    type : "function"
  06 <= : true : {}
    result : true
  07 ! : false : {}
    message : "Did not meet the not-array validator"
    reason : "some-failed"
    result : false
    subResult : []
      0 : {}
        message : "The sub-validator returns true in not-array check."
        result : true
        vIndex : 1
    type : "not-array"
  07 ! : true : {}
    result : true
  08 value : false : {}
    message : "Did not exact equal any value in the value-array"
    reason : "all-values-failed"
    result : false
    subResult : []
      0 : {}
        expected : 1
        message : "Target is not equal to the expected value."
        reason : "not-equal"
        result : false
        type : "exact-equal"
        vIndex : 1
      1 : {}
        expected : null
        message : "Target is not equal to the expected value."
        reason : "not-equal"
        result : false
        type : "exact-equal"
        vIndex : 2
    type : "value-array"
  08 value : true, 1 : {}
    result : true
  08 value : true, null : {}
    result : true
  09 or : false : {}
    message : "Did not exact equal any value in the value-array"
    reason : "all-values-failed"
    result : false
    subResult : []
      0 : {}
        expected : "array"
        message : "Target is not equal to the expected value."
        reason : "not-equal"
        result : false
        type : "exact-equal"
        vIndex : 1
      1 : {}
        expected : null
        message : "Target is not equal to the expected value."
        reason : "not-equal"
        result : false
        type : "exact-equal"
        vIndex : 2
    type : "value-array"
  09 or : true : {}
    result : true