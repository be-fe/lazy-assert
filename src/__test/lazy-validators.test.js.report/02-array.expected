{}
  01 or : "123", string | number : {}
    result : true
  01 or : 123, string | number : {}
    result : true
  01 or : null, string | number : {}
    message : "Did not match or-array"
    result : false
    target : null
    validator : []
      0 : "string"
      1 : "number"
  02 and : false, func error : {}
    index : 2
    message : "Did not meet all validators in and array"
    result : false
    subResult : {}
      funcInfo : undefined
      result : false
      target : 4
    subValidator : "_[[[function]]]_"
  02 and : false, type error : {}
    index : 1
    message : "Did not meet all validators in and array"
    result : false
    subResult : {}
      message : "Target value is not of the expected type"
      result : false
      target : "123"
      type : "number"
    subValidator : "number"
  02 and : true : {}
    result : true
  03 > : false : {}
    funcInfo : "Comparison : > 3"
    result : false
    target : 2
  03 > : true : {}
    result : true
  04 < : false : {}
    funcInfo : "Comparison : < 3"
    result : false
    target : 5
  04 < : true : {}
    result : true
  05 >= : false : {}
    funcInfo : "Comparison : > 3"
    result : false
    target : 2
  05 >= : true : {}
    result : true
  06 <= : false : {}
    funcInfo : "Comparison : <= 3"
    result : false
    target : 5
  06 <= : true : {}
    result : true
  07 ! : false : {}
    index : 1
    message : "Did not meet the not-array validator"
    result : false
    subValidator : "null"
    target : null
    type : "not array"
  07 ! : true : {}
    result : true
  08 value : false : {}
    message : "Did not exact equal any value in the value-array"
    result : false
    target : true
    validator : []
      0 : "value"
      1 : 1
      2 : null
  08 value : true, 1 : {}
    result : true
  08 value : true, null : {}
    result : true
  09 or : false : {}
    message : "Did not exact equal any value in the value-array"
    result : false
    target : true
    validator : []
      0 : "value"
      1 : "array"
      2 : null
  09 or : true : {}
    result : true